electricitymap/contrib/capacity_parsers/IRENA.py

```diff
@@ -0,0 +1,146 @@
+        f"ELECCAP_{url_year-1}.px",
+    ]
+    json_query = {
+        "query": [
+            {
+                "code": "Year",
+                "selection": {
+                    "filter": "item",
+                    "values": [target_datetime.strftime("%y")],
+                },
+            }
+        ],
+        "response": {"format": "json"},
+    }
+    data = None
+    for filename in filename_combinations:
+        url = base_url + filename
+
+        json_data = json.dumps(json_query)
+        r: Response = Session().post(url, data=json_data)
+        if r.status_code == 200:
+            data = r.json()
+        else:
+            continue
+    if not data:
+        raise ValueError(f"Could not fetch data for {target_datetime.year}")
+    return data["data"]
+
+
+def reallocate_capacity_mode(zone_key: ZoneKey, mode: int) -> dict:
+    if zone_key in SPECIFIC_MODE_MAPPING:
+        if mode in SPECIFIC_MODE_MAPPING[zone_key]:
+            return SPECIFIC_MODE_MAPPING[zone_key][mode]
+    else:
+        return IRENA_JSON_TO_MODE_MAPPING[mode]
+
+
+def get_capacity_data_for_all_zones(target_datetime: datetime):
+    data = get_data_from_url(target_datetime)
+    capacity_dict = {}
+    for item in data:
+        if pycountry.countries.get(alpha_3=item["key"][0]) is not None:
+            zone = pycountry.countries.get(alpha_3=item["key"][0]).alpha_2
+        else:
+            pass
+        mode: str = IRENA_JSON_TO_MODE_MAPPING[int(item["key"][1])]
+        value: float = round(float(item["values"][0]), 0)
+        datetime_value: datetime = datetime.strptime(item["key"][-1], "%y")
+
+        if zone not in capacity_dict:
+            zone_dict = {
+                mode: {
+                    "datetime": datetime_value,
+                    "value": value,
+                }
+            }
+            capacity_dict[zone] = zone_dict
+        else:
+            mode = reallocate_capacity_mode(zone, int(item["key"][1]))
+            if mode in capacity_dict[zone]:
+                zone_dict = capacity_dict[zone][mode]
+                capacity_dict[zone][mode]["value"] += value
+            else:
+                capacity_dict[zone] = {
+                    **capacity_dict[zone],
+                    **{
+                        mode: {
+                            "datetime": datetime_value,
+                            "value": value,
+                        }
+                    },
+                }
+    return capacity_dict
+
+
+def fetch_production_capacity(target_datetime: datetime, zone_key: ZoneKey) -> dict:
+    all_capacity = get_capacity_data_for_all_zones(target_datetime)
+    zone_capacity = all_capacity[zone_key]
+
+    if zone_capacity:
+        print(
+            f"Fetched capacity for {zone_key} in {target_datetime.year}: \n{zone_capacity}"
+        )
+        return zone_capacity
+    else:
+        raise ValueError(f"No capacity data for {zone_key} in {target_datetime.year}")
```

Comment:
Do we want the fetch capacity to crash if there is no capacity data? In the REE parser we used `logger.warning` to log it but continue with the process. Would it make sense to use the same approach across all capacity parsers?